### DISCUSSION TITLE
Which default sampler?

### DISCUSSION THREAD

**Discussion by DeLaRiva** [optuna/optuna#3804] [Q&A] [answered]
Hi,

which sampler is using Optuna as default sampler? Could you also please point me where I would find this information? I'm searching the docs for some while ago and it is very detailed (and tbh mostly above my head).
Thanks a lot!

**Comment by nzw0301** [comment, marked as answer]
It depends on the number of objectives. `TPESampler` is for a single objective and `NSGAIISampler` for multi-objectives. 
This information is provided in `sampler` argument's explanation on https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.create_study.html#optuna.study.create_study.

**Reply by Koen1999** [reply]
Does this advice still stand at the beginning of 2024, a while after `NSGAIIISampler` was introduced in v3.20?

**Reply by nzw0301** [reply]
The default samplers are same depending on the number of objectives.

**Reply by nzw0301** [reply]
I'm not familiar with NSGA 3 but it does not always outperform NSGA 2 according to the abstract of https://ieeexplore.ieee.org/document/7744174, so the optimal sampler will be problem dependent.

**Comment by DeLaRiva** [comment]
Ok, thank you!
Indeed, I didn't look there but I guess I read every other page about the sampler themselves :)

Do you know any advantages or disadvantages or specific purposes of the other optimizers? E.g. is one better suited for continuous spaces/variables or for sparse landscapes with multiple peaks and so on?

**Reply by nzw0301** [reply]
https://optuna.readthedocs.io/en/latest/reference/samplers/index.html might be a useful page for you :)

**Reply by DeLaRiva** [reply]
Very true, thanks a lot! I missed that.. but btw, I cannot see the BoTorchSampler in the sublist of the samplers, only in optuna.integration. Is this an external package I have to bind in?

Do you mind helping out more for some basic questions? E.g., what does "log" mean in trial.suggest_float()? The [documentation ](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.Trial.html#optuna.trial.Trial.suggest_float) tells

_A flag to sample the value from the log domain or not. If log is true, the value is sampled from the range in the log domain. Otherwise, the value is sampled from the range in the linear domain._

I understand this simply as taking the log() of a value or not. But why or how would this matter?
